File: src/main/java/org/micoli/php/ui/components/tasks/tree/TreeCellRenderer.java
package org.micoli.php.ui.components.tasks.tree;

import com.intellij.ide.util.treeView.NodeRenderer;
import javax.swing.*;
import org.jetbrains.annotations.NotNull;

public class TreeCellRenderer extends NodeRenderer {
    @Override
    public void customizeCellRenderer(
            @NotNull JTree tree,
            Object value,
            boolean selected,
            boolean expanded,
            boolean leaf,
            int row,
            boolean hasFocus) {

        if (value instanceof PathNode pathNode) {
            this.append(pathNode.getLabel());
            return;
        }

        if (value instanceof DynamicTreeNode dynamicTreeNode) {
            this.setIcon(dynamicTreeNode.getIcon());
            this.append(dynamicTreeNode.getLabel());
            if (dynamicTreeNode.getTooltip() != null) {
                this.setToolTipText(dynamicTreeNode.getTooltip());
            }
            return;
        }

        super.customizeCellRenderer(tree, value, selected, expanded, leaf, row, hasFocus);
    }
}

File: src/main/kotlin/org/micoli/php/ui/components/tasks/tree/tasks/TreeCellRenderer.kt
package org.micoli.php.ui.components.tasks.tree.tasks

import com.intellij.ide.util.treeView.NodeRenderer
import javax.swing.JTree

class TreeCellRenderer : NodeRenderer() {
    override fun customizeCellRenderer(
        tree: JTree,
        value: Any?,
        selected: Boolean,
        expanded: Boolean,
        leaf: Boolean,
        row: Int,
        hasFocus: Boolean,
    ) {
        if (value is PathNode) {
            this.append(value.getLabel())
            return
        }

        if (value is DynamicTreeNode) {
            this.setIcon(value.icon)
            this.append(value.getLabel())
            this.setToolTipText(value.getTooltip())
            return
        }

        super.customizeCellRenderer(tree, value, selected, expanded, leaf, row, hasFocus)
    }
}
